cmd/kube-proxy/app/server.go

```diff
@@ -904,11 +904,21 @@ func (s *ProxyServer) Run() error {
 		return err
 	}
 
+	// despite there are independent informers for EndpointSlices and Services, for backwards compatibility we
+	// filter both objects with the same labels, despite noProxyName is meant to be used only in Services and
+	// noHeadlessEndpoints only for EndpointSlices
 	labelSelector := labels.NewSelector()
 	labelSelector = labelSelector.Add(*noProxyName, *noHeadlessEndpoints)
 
-	// Make informers that filter out objects that want a non-default service proxy.
-	informerFactory := informers.NewSharedInformerFactoryWithOptions(s.Client, s.Config.ConfigSyncPeriod.Duration,
+	// Create informer for Services, filter out the ones that want a non-default service proxy or a Headless Service.
+	serviceInformer := informers.NewSharedInformerFactoryWithOptions(s.Client, s.Config.ConfigSyncPeriod.Duration,
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			options.LabelSelector = labelSelector.String()
+			options.FieldSelector = fields.OneTermNotEqualSelector("spec.clusterIP", v1.ClusterIPNone).String()
```

Comment:
this will fail against an old apiserver :/

Reply:
this is covered on our skew policies though https://kubernetes.io/releases/version-skew-policy/